### PR TITLE
Move FIE code to fie.js to fix amp4ads-v0.js size regression

### LINKED ISSUE #23487: amp4ads-v0.js size regression
The bundle size of amp4ads-v0.js went back to 285K.

Caused by #23153 
in specific this line:
https://github.com/ampproject/amphtml/blob/506b47f2298a2e9b9fca901d3c4aa090c604ad9b/src/service/extensions-impl.js#L469

/cc @dvoytenko

### PR DESCRIPTION
**Pull request by lannka**
Fixes #23487

- Moved a couple of embed/childwin only methods from extensions-impl.js to fie.js
- Replaced sinon.mock with sinon.stub in some of the tests
- Achieved some extra bytes saving for amp4ads-v0.js by reducing the size of extensions-impl.js

amp4ads-v0.js is now 262KB

### CODE REVIEW

**Comment by lannka** [comment]
@choumx & @jridgewell as @dvoytenko is out

**Comment by choumx** [comment]
> Δ -0.57KB

Nice. Was this a recent size regression?

File: src/friendly-iframe-embed.js [hunk 1]
```diff
@@ -228,13 +250,15 @@ export function installFriendlyIframeEmbed(
...
-      extensions.installExtensionsInFie(
+      embed.installExtensionsInFie(
+        extensions,
         ampdoc,
         spec.extensionIds || [],
         opt_preinstallCallback
       );
     } else {
-      extensions.installExtensionsInChildWindow(
+      embed.installExtensionsInChildWindow(
```
Comment (choumx) [inline src/friendly-iframe-embed.js]: You can probably change these methods to `@private` now that they're only accessed from the same file.

Comment (lannka) [reply src/friendly-iframe-embed.js]: still used in tests though. I marked them @visibleForTesting . although I guess that annotation is currently a no-op other than documentation purpose?

**Comment by lannka** [comment]
woops, circular dependencies:
```
src/service/core-services.js -> src/service/viewport/viewport-impl.js -> src/friendly-iframe-embed.js -> src/service/core-services.js
```

**Comment by lannka** [comment]
Had a temporary hack to work around the circular dependencies. will have a follow up PR to clear separate the installation & retrieval of a friendlyIframeEmbed.

**Pull request merged**